src/sound/soundbase.cpp

```diff
@@ -232,98 +221,156 @@ QVector<QString> CSoundBase::LoadAndInitializeFirstValidDriver ( const bool bOpe
 void CSoundBase::ParseCommandLineArgument ( const QString& strMIDISetup )
 {
-    int iMIDIOffsetFader = 70; // Behringer X-TOUCH: offset of 0x46
-
-    // parse the server info string according to definition: there is
-    // the legacy definition with just one or two numbers that only
-    // provides a definition for the controller offset of the level
-    // controllers (default 70 for the sake of Behringer X-Touch)
-    // [MIDI channel];[offset for level]
-    //
-    // The more verbose new form is a sequence of offsets for various
-    // controllers: at the current point, 'f', 'p', 's', and 'm' are
-    // parsed for fader, pan, solo, mute controllers respectively.
-    // However, at the current point of time only 'f' and 'p'
-    // controllers are actually implemented.  The syntax for a Korg
-    // nanoKONTROL2 with 8 fader controllers starting at offset 0 and
-    // 8 pan controllers starting at offset 16 would be
-    //
-    // [MIDI channel];f0*8;p16*8
-    //
-    // Namely a sequence of letters indicating the kind of controller,
-    // followed by the offset of the first such controller, followed
-    // by * and a count for number of controllers (if more than 1)
-    if ( !strMIDISetup.isEmpty() )
+    if ( strMIDISetup.isEmpty() )
     {
-        // split the different parameter strings
-        const QStringList slMIDIParams = strMIDISetup.split ( ";" );
+        // should be caught in main.cpp
+        return;
+    }
+
+    // Parse the --ctrlmidich string.  There are two formats.
+    // Default to the legacy kind of specifying the fader controller offset
+    // without an indication of the count of controllers.
+    bool bSimple = true;
 
-        // [MIDI channel]
-        if ( slMIDIParams.count() >= 1 )
+    // split the different parameter strings
+    const QStringList slMIDIParams = strMIDISetup.split ( ";" );
+    int               iNumParams   = slMIDIParams.count();
+
+    if ( iNumParams >= 1 )
+    {
+        bool bChOK = false;
+        int  i     = slMIDIParams[0].toUInt ( &bChOK );
+        if ( bChOK )
         {
-            iCtrlMIDIChannel = slMIDIParams[0].toUInt();
+            // [MIDI channel] supplied (else use default)
+            iCtrlMIDIChannel = i;
         }
+        else
+        {
+            // iCtrlMIDIChannel == INVALID_MIDI_CH, so no point continuing
+            return;
+        }
+    }
 
-        bool bSimple = true; // Indicates the legacy kind of specifying
-                             // the fader controller offset without an
-                             // indication of the count of controllers
+    // Use Behringer X-TOUCH as default offset of 0x46
+    int iMIDIOffsetFader = 70;
+    if ( iNumParams >= 2 )
+    {
+        // if there is a second parameter that can be parsed as a number,
+        // we have the legacy specification of controllers.
+        int i = slMIDIParams[1].toUInt ( &bSimple );
```

Comment:
Can toUInt fail? What happens then?

Reply:
Also UInt != int?

Reply:
`toUInt` sets the boolean argument passed by reference to true or false, depending on success.

`i` should be a number from 0 to 127.  Of course, someone _could_ type a really stupid value in that would give a really weird result...  The `toUInt` is to prevent parsing of a valid negative value, though, as that could cause issues.